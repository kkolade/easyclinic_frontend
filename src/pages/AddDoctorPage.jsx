import { useDocumentTitle } from '@mantine/hooks';

import AppShell from 'components/AppShell';

const AddDoctorPage = () => {
  useDocumentTitle('Add Doctor - EasyClinic');

  return <AppShell>AddDoctorPage</AppShell>;
};

export default AddDoctorPage;
