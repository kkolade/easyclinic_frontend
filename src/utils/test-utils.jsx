/* eslint-disable import/no-extraneous-dependencies */

import { MantineProvider } from '@mantine/core';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import theme from '../theme';

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    <MantineProvider withNormalizeCSS withGlobalStyles theme={theme}>
      <Router>{children}</Router>
    </MantineProvider>
  </Provider>
);

const customRender = (ui, options) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
