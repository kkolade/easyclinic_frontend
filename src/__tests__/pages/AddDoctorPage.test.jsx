import { render } from 'test-utils';

import AddDoctorPage from 'pages/AddDoctorPage';

describe('AddDoctorPage', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<AddDoctorPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
