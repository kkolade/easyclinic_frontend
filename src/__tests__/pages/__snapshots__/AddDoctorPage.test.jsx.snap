// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDoctorPage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mantine-AppShell-root mantine-1udmbix"
  >
    <div
      class="mantine-hr6aor mantine-AppShell-body"
    >
      <nav
        class="mantine-Navbar-root mantine-13ge6fs"
        data-hidden="true"
      >
        <div
          class="mantine-qyxvtq"
        >
          <div
            class="mantine-Image-root mantine-lgo8bz"
            style="width: 12.5rem;"
          >
            <figure
              class="mantine-qenwvq mantine-Image-figure"
            >
              <div
                class="mantine-1iugybl mantine-Image-imageWrapper"
              >
                <img
                  alt="EasyClinic Logo"
                  class="mantine-1i6l9bv mantine-Image-image"
                  src="/logo.png"
                  style="object-fit: cover; width: 12.5rem; height: auto;"
                />
              </div>
            </figure>
          </div>
        </div>
        <div
          class="mantine-hzudtq"
        >
          <div>
            <a
              class="mantine-UnstyledButton-root mantine-NavLink-root mantine-165mgsf"
              data-active="true"
              href="/"
            >
              <span
                class="mantine-1l4mhad mantine-NavLink-icon"
              >
                <svg
                  class="tabler-icon tabler-icon-home"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12l-2 0l9 -9l9 9l-2 0"
                  />
                  <path
                    d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
                  />
                  <path
                    d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
                  />
                </svg>
              </span>
              <span
                class="mantine-dstkpx mantine-NavLink-body"
              >
                <span
                  class="mantine-Text-root mantine-NavLink-label mantine-1sdv04i"
                >
                  Home
                </span>
                <span
                  class="mantine-Text-root mantine-NavLink-description mantine-h4oguy"
                  data-active="true"
                />
              </span>
            </a>
            <div
              aria-hidden="true"
              class="mantine-1avyp1d"
              style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
            >
              <div
                style="opacity: 0; transition: opacity 200ms ease;"
              >
                <div
                  class="mantine-1c04k7q mantine-NavLink-children"
                />
              </div>
            </div>
            <a
              class="mantine-UnstyledButton-root mantine-NavLink-root mantine-165mgsf"
              href="/signin"
            >
              <span
                class="mantine-1l4mhad mantine-NavLink-icon"
              >
                <svg
                  class="tabler-icon tabler-icon-login"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 8v-2a2 2 0 0 0 -2 -2h-7a2 2 0 0 0 -2 2v12a2 2 0 0 0 2 2h7a2 2 0 0 0 2 -2v-2"
                  />
                  <path
                    d="M20 12h-13l3 -3m0 6l-3 -3"
                  />
                </svg>
              </span>
              <span
                class="mantine-dstkpx mantine-NavLink-body"
              >
                <span
                  class="mantine-Text-root mantine-NavLink-label mantine-1sdv04i"
                >
                  Sign in
                </span>
                <span
                  class="mantine-Text-root mantine-NavLink-description mantine-h4oguy"
                />
              </span>
            </a>
            <div
              aria-hidden="true"
              class="mantine-1avyp1d"
              style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
            >
              <div
                style="opacity: 0; transition: opacity 200ms ease;"
              >
                <div
                  class="mantine-1c04k7q mantine-NavLink-children"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mantine-khtkeg"
        >
          <div
            class="mantine-Text-root mantine-1p5v0xp"
          >
            Â© 2023 EasyClinic
          </div>
        </div>
      </nav>
      <main
        class="mantine-bzbzc1 mantine-AppShell-main"
      >
        <div
          class="mantine-cffs0p"
        >
          <div
            class="mantine-1ycigyy"
          >
            <h3
              class="mantine-Text-root mantine-Title-root mantine-jgtvnu"
            >
              Create a doctor
            </h3>
          </div>
          <form
            class="mantine-gr49qu"
          >
            <div
              class="mantine-1x8qhrp"
            >
              <div
                class="mantine-InputWrapper-root mantine-TextInput-root mantine-12pjkll"
              >
                <label
                  class="mantine-InputWrapper-label mantine-TextInput-label mantine-1fzet7j"
                  for="mantine-yl6n2yqtm"
                  id="mantine-yl6n2yqtm-label"
                >
                  Doctor name
                </label>
                <div
                  class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-1v7s5f8"
                >
                  <input
                    aria-invalid="false"
                    class="mantine-Input-input mantine-TextInput-input mantine-1153slh"
                    id="mantine-yl6n2yqtm"
                    placeholder="Enter doctor name"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mantine-InputWrapper-root mantine-TextInput-root mantine-12pjkll"
              >
                <label
                  class="mantine-InputWrapper-label mantine-TextInput-label mantine-1fzet7j"
                  for="mantine-qf25y0ts4"
                  id="mantine-qf25y0ts4-label"
                >
                  Profile photo
                </label>
                <div
                  class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-1v7s5f8"
                >
                  <input
                    aria-invalid="false"
                    class="mantine-Input-input mantine-TextInput-input mantine-1153slh"
                    id="mantine-qf25y0ts4"
                    placeholder="Enter image URL"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mantine-InputWrapper-root mantine-NumberInput-root mantine-12pjkll"
              >
                <label
                  class="mantine-InputWrapper-label mantine-NumberInput-label mantine-1fzet7j"
                  for="mantine-w07f5byz0"
                  id="mantine-w07f5byz0-label"
                >
                  Experience (years)
                </label>
                <div
                  class="mantine-Input-wrapper mantine-NumberInput-wrapper mantine-1v7s5f8"
                >
                  <input
                    aria-invalid="false"
                    class="mantine-Input-input mantine-NumberInput-input mantine-1p0jw6r"
                    id="mantine-w07f5byz0"
                    inputmode="numeric"
                    required=""
                    step="1"
                    type="text"
                    value="5"
                  />
                  <div
                    class="mantine-unhde mantine-Input-rightSection mantine-NumberInput-rightSection"
                  >
                    <div
                      class="mantine-1s0kk1f mantine-NumberInput-rightSection"
                    >
                      <button
                        aria-hidden="true"
                        class="mantine-NumberInput-control mantine-NumberInput-controlUp mantine-x0i9fi"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          fill="none"
                          style="width: 0.875rem; height: 0.875rem; transform: rotate(180deg);"
                          viewBox="0 0 15 15"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                      <button
                        aria-hidden="true"
                        class="mantine-NumberInput-control mantine-NumberInput-controlDown mantine-1g3thxc"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          fill="none"
                          style="width: 0.875rem; height: 0.875rem;"
                          viewBox="0 0 15 15"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="mantine-InputWrapper-root mantine-TextInput-root mantine-12pjkll"
              >
                <label
                  class="mantine-InputWrapper-label mantine-TextInput-label mantine-1fzet7j"
                  for="mantine-e8qibudw9"
                  id="mantine-e8qibudw9-label"
                >
                  Bio
                </label>
                <div
                  class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-1v7s5f8"
                >
                  <input
                    aria-invalid="false"
                    class="mantine-Input-input mantine-TextInput-input mantine-nb24ab"
                    id="mantine-e8qibudw9"
                    placeholder="Share your professional background, interests, and accomplishments."
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-ulxcsg"
                data-button="true"
                type="submit"
              >
                <div
                  class="mantine-1wpc1xj mantine-Button-inner"
                >
                  <span
                    class="mantine-1ryt1ht mantine-Button-label"
                  >
                    Add doctor
                  </span>
                </div>
              </button>
            </div>
          </form>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
